docs/api/context-bridge.md

```diff
@@ -0,0 +1,127 @@
+# contextBridge
+
+> Create a safe, bi-directional, synchronous bridge across isolated contexts
+
+Process: [Renderer](../glossary.md#renderer-process)
+
+An example of exposing a basic API to a renderer from an isolated preload script is given below:
+
+```javascript
+// Preload (Isolated World)
+const { contextBridge, ipcRenderer } = require('electron')
+
+contextBridge.exposeInMainWorld(
+  'electron',
+  {
+    doThing: () => ipcRenderer.send('do-a-thing')
+  }
+)
+```
+
+```javascript
+// Renderer (Main World)
+
+window.electron.doThing()
+```
+
+## Glossary
+
+### Main World
+
+The "Main World" is the javascript context that your main renderer code runs in.  By default the page you load in your renderer
+executes code in this world.
+
+### Isolated World
+
+When `contextIsolation` is enabled in your `webPreferences` your `preload` scripts run in an "Isolated World".  You can read more about
+context isolation and what it affects in the [BrowserWindow](browser-window.md) docs.
+
+## Methods
+
+The `contextBridge` module has the following instance methods:
+
+### `contextBridge.exposeInMainWorld(apiKey, api)`
+
+* `apiKey` String - The key to inject the API onto `window` with.  The API will be accessible on `window[apiKey]`.
+* `api` Record<String, any> - Your API object, more information on what this API can be and how it works is available below.
+
+## Usage
+
+### API Objects
+
+The `api` object provided to [`exposeInMainWorld`](#contextbridgeexposeInMainWorldapikey-api-options) must be an object
+whose keys are strings and values are a `Function`, `String`, `Number`, `Array`, `Boolean` or another nested object that meets the same conditions.
+
+`Function` values are proxied to the other context and all other values are **copied** and **frozen**.  I.e. Any data / primtives sent in
+the API object become immutable and updates on either side of the bridge do not result in an update on the other side.
```

Comment:
Proxy'ing getters and setters is infinitely more complicated and may result in Hidden Cost issues where an object gets passed around until it isn't clear it's a magical proxy object.  Then when folks use `a.b = myBigObject` it's quite slow because it gets serialized twice.

Opinionated but IMO it's better to remove that chance of the Hidden Cost.

Reply:
> serialized twice

I'm not sure I get what would be serialized twice unexpectedly here?

I was imagining a read-only proxy layer, so that updates in the isolated world are reflected in the main world. But perhaps that's enough of an edge case that we should go with the simpler option for now :+1: